src/libraries/System.Net.Quic/src/System/Net/Quic/QuicStream.cs

```diff
@@ -120,6 +120,12 @@ public sealed partial class QuicStream
     private long _id = -1;
     private readonly QuicStreamType _type;
 
+    /// <summary>
+    /// Provided via <see cref="StartAsync(Action{QuicStreamType}, CancellationToken)" /> from <see cref="QuicConnection" /> so that <see cref="QuicStream"/> can decrement its available stream count field.
+    /// When <see cref="HandleEventStartComplete(ref START_COMPLETE_DATA)">START_COMPLETE</see> arrives it gets invoked and unset back to <c>null</c> to not to hold any unintended reference to <see cref="QuicConnection"/>.
+    /// </summary>
+    private Action<QuicStreamType>? _decrementAvailableStreamCount;
```

Comment:
Yep, it is. The intent here was to express that we need the link only until the stream opening (=reserving the Stream ID) is confirmed by MsQuic. After that, the delegate gets nulled, destroying the link. But let's wait for the API review of https://github.com/dotnet/runtime/issues/101534 and if we stick with the increments, I'll try to push this to the MsQuic layer as you suggested and this will moot.

Reply:
I'm keeping this as-is for now. I have added the `Action` delegate caching in `QuicConnection`. I do plan to push for the change in msquic, but I'll not hold this PR on it.